src/theme/src/hooks/usePaneAppearance.js

```diff
@@ -0,0 +1,18 @@
+import { useMemo } from 'react'
+import useStyleConfig from '../../../hooks/use-style-config'
+import getPaneStyles from '../components/pane'
+import useTheme from '../useTheme'
+
+const pseudoSelectors = {
+  _hover: '&:hover',
+  _active: '&:active'
+}
+
+function usePaneAppearance(modifiers, internalStyles) {
+  const theme = useTheme()
+  const paneStyles = useMemo(() => getPaneStyles(theme), [theme])
+
+  return useStyleConfig(paneStyles, modifiers, pseudoSelectors, internalStyles)
+}
```

Comment:
I think this can end up getting pulled into `useStyleConfig`, but would require a key lookup on the `components/*` object to find the `getFooStyles` function. Then all of these `useFooAppearance` hooks go away.

Reply:
Still think this abstraction makes sense because each internal `pseudoSelectors` are bound to the `useXAppearance()` hook, and not exposed anywhere else. 

Unless I'm missing something?

Reply:
That's correct, but it may make a ton more sense to move all of this out of the theme directory and have it colocated next to the component that uses it. If `useComponentAppearance` hook goes away, then the only remaining code (that could sit next to the component itself) is `pseudeoSelectors` + the style config fn.

Also I would love a world where the pseudoSelectors are actually standardized and not ad-hoc per component. Not too sure if that's possible yet.